spec/system/user_authorization_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+def cas_login
+  fill_in 'username', with: 'john.doe@email.com'
+  fill_in 'password', with: 'any password'
+  click_button 'Login'
+end
+
+describe 'User authorization', type: :system, js: true do
+  it 'visits the frontend as editor' do
```

Comment:
This can be simpler with this structure:

when "user is editor"
  it "can access the frontend"
  it "can't access the backend"
end

when "user is admin"
  it "can access the frontend"
  it "can access the backend"
end